File: src/new_test.go
package vibe

import "testing"

func TestSanitizeTaskName(t *testing.T) {
	tests := []struct {
		input    string
		expected string
	}{
		{"simple", "simple"},
		{"with-dash", "with-dash"},
		{"with_underscore", "with_underscore"},
		{"CamelCase", "CamelCase"},
		{"with spaces", "with-spaces"},
		{"with/slashes", "with-slashes"},
		{"special!@#chars", "special-chars"},
		{"  trimmed  ", "trimmed"},
		{"---leading-trailing---", "leading-trailing"},
		{"multiple   spaces", "multiple-spaces"},
		{"日本語", ""},
		{"", ""},
		{"123", "123"},
		{"a1b2c3", "a1b2c3"},
	}

	for _, tt := range tests {
		t.Run(tt.input, func(t *testing.T) {
			result := sanitizeTaskName(tt.input)
			if result != tt.expected {
				t.Errorf("sanitizeTaskName(%q) = %q, want %q", tt.input, result, tt.expected)
			}
		})
	}
}

File: src/new.test.ts
import { describe, expect, test } from "bun:test";
import { sanitizeTaskName } from "./new";

describe("sanitizeTaskName", () => {
  test.each([
    { input: "simple", expected: "simple" },
    { input: "with-dash", expected: "with-dash" },
    { input: "with_underscore", expected: "with_underscore" },
    { input: "CamelCase", expected: "CamelCase" },
    { input: "with spaces", expected: "with-spaces" },
    { input: "with/slashes", expected: "with-slashes" },
    { input: "special!@#chars", expected: "special-chars" },
    { input: "  trimmed  ", expected: "trimmed" },
    { input: "---leading-trailing---", expected: "leading-trailing" },
    { input: "multiple   spaces", expected: "multiple-spaces" },
    { input: "日本語", expected: "" },
    { input: "", expected: "" },
    { input: "123", expected: "123" },
    { input: "a1b2c3", expected: "a1b2c3" },
  ])("sanitizeTaskName($input) = $expected", ({ input, expected }) => {
    expect(sanitizeTaskName(input)).toBe(expected);
  });
});
